src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -549,21 +550,34 @@ def update(self, area):
             area, self._price_energy_day,
             use_last_past_market=not ConstSettings.GeneralSettings.KEEP_PAST_MARKETS
         )
-        for node, trade_rates in self._price_energy_day.items():
-            if node.name not in self.csv_output:
-                self.csv_output[node.name] = {
+        latest_output_area_names = self.gather_trade_rates(
+            area, {},
+            use_last_past_market=not ConstSettings.GeneralSettings.KEEP_PAST_MARKETS
+        )
+
+        # Calculate aggregated results for all markets
+        self._convert_output_format(self._price_energy_day, self.csv_output, self.redis_output)
+
+        # Calculate results for one market
+        self._convert_output_format(latest_output_area_names, {}, self.latest_output)
+
+        return self.csv_output, self.redis_output
```

Comment:
AFAIK, wherever "update" is used, the return values are not used. So you could remove this line, right?

Reply:
Right, deleting this